Echo.Process/Session/SessionDataTypeResolve.cs

```diff
@@ -0,0 +1,79 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using LanguageExt;
+using static LanguageExt.Prelude;
+using static Echo.Process;
+
+namespace Echo.Session
+{
+    public static class SessionDataTypeResolve
+    {
+        static Map<string, Option<Type>> sessionDataTypeValid = Map<string, Option<Type>>();
```

Comment:
Why is this an `Option`?

Reply:
The existence of the item in the `Map` is its optionality

Reply:
As discussed, if we fail to resolve a typename, we want to store that information as none in the map so the next time same typename is requested, we don't have to resolve it again.